ui/button.mod/button.js

```diff
@@ -1,485 +1,498 @@
+        this._pressComposer.addEventListener("pressCancel", this, false);
+
+        // FIXME: @benoit: we should maybe have a flag for this kind of event.
+        // can be tricky with the event delegation for example if we don't add it.
+        // same issue for: the pressComposer and the translate composer.
+        this._pressComposer.addEventListener("longPress", this, false);
+    }
+
+    /**
+     * Removes event listeners from the button.
+     * @private
+     */
+    _removeEventListeners() {
+        this._pressComposer.removeEventListener("press", this, false);
+        this._pressComposer.removeEventListener("pressCancel", this, false);
+        this._pressComposer.removeEventListener("longPress", this, false);
+    }
+
+    /**
+     * Applies the current image position's styling by updating CSS classes
+     * @private
+     * @see Button.IMAGE_POSITIONS
+     */
+    _applyImagePositionStyles() {
+        this._removeClassListTokens(...Object.values(Button.IMAGE_POSITIONS));
+        this.classList.add(this.imagePosition);
+    }
+
+    /**
+     * Applies the current orientation's styling by updating CSS classes
+     * @private
+     * @see Button.ORIENTATIONS
+     */
+    _applyOrientationStyles() {
+        this._removeClassListTokens(...Object.values(Button.ORIENTATIONS));
+        this.classList.add(this.orientation);
+    }
+
+    // FIXME: Remove this method when the classList's remove method is fixed!
+    // Our implementation doesn't support multiple arguments
+    // https://dom.spec.whatwg.org/#dom-domtokenlist-remove
+    _removeClassListTokens(...tokens) {
+        for (const token of tokens) {
+            this.classList.remove(token);
+        }
+    }
 });
 
-Button.addAttributes( /** @lends module:"mod/ui/native/button.mod".Button# */{
+Button.addAttributes({
+    /** @lends module:"mod/ui/native/button.mod".Button# */
 
-/**
-    The URL to which the form data will be sumbitted.
-    @type {string}
-    @default null
-*/
+    /**
+     * The URL to which the form data will be sumbitted.
+     * @type {string}
+     * @default null
+     */
     formaction: null,
 
-/**
-    The content type used to submit the form to the server.
-    @type {string}
-    @default null
-*/
+    /**
+     * The content type used to submit the form to the server.
+     * @type {string}
+     * @default null
+     */
     formenctype: null,
 
-/**
-    The HTTP method used to submit the form.
-    @type {string}
-    @default null
-*/
+    /**
+     * The HTTP method used to submit the form.
+     * @type {string}
+     * @default null
+     */
     formmethod: null,
 
-/**
-    Indicates if the form should be validated upon submission.
-    @type {boolean}
-    @default null
-*/
-    formnovalidate: {dataType: 'boolean'},
+    /**
+     * Indicates if the form should be validated upon submission.
+     * @type {boolean}
+     * @default null
+     */
+    formnovalidate: { dataType: "boolean" },
```

Comment:
Do those still apply if a button can'y be used with an input type="button" ? We should consider that and answer it

Reply:
yup, https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#formnovalidate